### PR TITLE
Trying out decorators for common checks

### PR DESCRIPTION
**Pull request by richardjgowers**
So mostly a stylistic question to @MDAnalysis/coredevs , does this look like an improvement?  We're often doing these checks, and this way allows us to treat them in a more uniform way (same error messages and Error raised for a given check).

### CODE REVIEW

File: package/MDAnalysis/core/groups.py [hunk 1]
```diff
@@ -14,6 +14,28 @@
 from . import topologyobjects
 
 
+def same_universe_check(f):
+    # Check a two object method occurs between objects in same Universe
+    @functools.wraps(f)
+    def wrapper(self, other):
+        if not self.universe is other.universe:
```
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py]: stylistically, it is nicer to say
```python
if self.universe is not other universe:
```

**Comment by orbeckst** [comment]
Looks like a good idea, especially if it cuts down on duplicated code.

**Comment by kain88-de** [comment]
I rather have them as normal short functions then decorators. That would still remove duplicate code and I could use the small functions in my own scripts easier. Functions also allow us more freedom in the app design while decorators expect a certain api

**Comment by jbarnoud** [comment]
To bounce on @kain88-de, we could have them as function and have a very light decorator that wrap these functions. We could also have decorators that wrap all the tests needed. For instance a `@operator_check`.

**Comment by kain88-de** [comment]
I generally really like the idea. As a matter of personal taste though I prefer to use small functions `is_same_universe` instead of decorators because I think they are easier to understand then decorators  (also for people new to python).

**Comment by richardjgowers** [comment]
Ok I think I'm going with @kain88-de 's idea for small functions for stuff like this

**Pull request closed without merging**